packages/material-ui/src/FormControlLabel/FormControlLabel.d.ts

```diff
@@ -13,7 +13,7 @@ export interface FormControlLabelProps
   inputRef?: React.Ref<any>;
   label: React.ReactNode;
   name?: string;
-  onChange?: (event: React.ChangeEvent<{}>, checked: boolean) => void;
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>, checked: boolean) => void;
```

Comment:
This depends on the component passed to the `control` prop. Are you sure this is always a native element?

Reply:
Yeah, I did not realize how general `FormControlLabel` could be, this is a stronger assumption than the one made for `RadioGroup`. But since the docs present this component as "drop in replacement of the Radio, Switch and Checkbox", do you think loosening up this definition to `React.ChangeEvent<any>` would be bad? The users could access `event.target.value` or `event.target.checked` (which is what the API docs suggest) without any additional burden and they can provide a stricter type definition if they prefer.

Alternatively, we could leave it as it is and add the code snippet you posted in the TypeScript guide? Just to make sure no one is confused by this particular case 🙂

Reply:
@danielasy 
So this is just opinionated but I believe that typescript is more harmful if used with `any` than useful. Therefore we should avoid it by default and let the user opt into dangerous behavior.

The good thing is that this encourages users to explicitly declare their callback argument types which will help transition into #11731 when we roll it out.

I would like an addition the the typescript guide explaining how to loosen up the checks in callbacks.